tests/integration.rs

```diff
@@ -97,7 +97,11 @@ fn main() -> Result<()> {
     config.stdout_filter(r#"exit status: 0xc0000409"#, "signal: 6 (SIGABRT)");
     config.filter("\"--target=[^\"]+\"", "");
 
-    let text = ui_test::status_emitter::Text::from(args.format);
+    let emitter: Box<dyn status_emitter::StatusEmitter> = match args.format {
+        Format::JSON => Box::new(status_emitter::JSON),
+        Format::Pretty => Box::new(status_emitter::Text::verbose()),
+        Format::Terse => Box::new(status_emitter::Text::quiet()),
+    };
```

Comment:
Add a method to `Format` or a free function to `status_emitter` to do this, and use it in all the duplicated places

Reply:
I first tried a `Format` implementation, but both the use in `integrations.rs` and `lib.rs` did not like `status_emitter::from(args.format)`. I suspect that the path I used was missing something, but my Rust-fu is not strong enough.

A free function did the job.